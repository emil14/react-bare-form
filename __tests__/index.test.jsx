import renderer from 'react-test-renderer'
import React from 'react'
import Form from '../src'

test('Form renders with attributes', () => {
  const formProps = {
    name: 'myForm',
    id: 'myForm',
    className: 'bare-form'
  }
  const tree = renderer
    .create(<Form {...formProps}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('Form renders its children', () => {
  const formProps = {
    name: 'myForm',
    content: [
      {
        name: 'mySelect',
        element: 'select',
        options: [
          {
            label: 'First option',
            value: 0,
          },
          {
            label: 'Second option',
            value: 1,
          }
        ]
      },
      {
        name: 'myInput',
        element: 'input',
        label: 'Password:',
        type: 'password'
      }
    ]
  }
  const tree = renderer
    .create(<Form {...formProps}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
