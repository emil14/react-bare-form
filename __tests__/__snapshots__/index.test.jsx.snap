// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders its children 1`] = `
<form
  name="myForm"
>
  <select
    className="form__select form__select--mySelect"
    name="mySelect"
  >
    <option
      label="First option"
      value={0}
    >
      First option
    </option>
    <option
      label="Second option"
      value={1}
    >
      Second option
    </option>
  </select>
  <label
    for="myInput"
  >
    Password:
  </label>
  <input
    className="form__input form__input--myInput"
    name="myInput"
    type="password"
  />
</form>
`;

exports[`Form renders with attributes 1`] = `
<form
  className="bare-form"
  id="myForm"
  name="myForm"
/>
`;
